main/src/ca/uwaterloo/flix/language/phase/Parser2.scala

```diff
@@ -879,8 +879,8 @@ object Parser2 {
 
     private def moduleDecl(mark: Mark.Opened, nestingLevel: Int = 0)(implicit s: State): Mark.Closed = {
       assert(at(TokenKind.KeywordMod))
-      expect(TokenKind.KeywordMod, SyntacticContext.Decl.Module)
-      name(NAME_MODULE, allowQualified = true, context = SyntacticContext.Decl.Module)
+      expect(TokenKind.KeywordMod, SyntacticContext.Decl.ModuleHeader)
+      name(NAME_MODULE, allowQualified = true, context = SyntacticContext.Decl.ModuleHeader)
       expect(TokenKind.CurlyL, SyntacticContext.Decl.Module)
```

Comment:
It seems to me it would be more appropriate to have:

SyntacticContext.Name
SyntacticContext.Keyword

Reply:
That makes sense. In all but a few cases, we don't care *what* kind of "header" we're in anyways. There might still be a reason to have a special context for the headers where "where" and "with" suggestions make sense. I know we decided to drop them suggestion support for them, but I think we might be able support them that way

Reply:
There is no such thing as a header. It is an abstraction you have invented.

Reply:
I'm aware that there's nothing named header in the code (except for what I added before). It's just a term I've been using to refer to a specific part of declarations. Most declarations have some kind of body or block and something that comes before that. For a module there's the name. For a function, there's the signature. For trait, there's name and optionally the with clause. For instances there's the name of the instance and what it's being implemented on. This part before the body or block, that's what I call the header. I didn't know what else to call it. A signature? But that's only what you'd call it for a function, right? I don't know if there is some other common term for it